workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs

```diff
@@ -90,6 +89,24 @@ private static void LaunchAndroid(bool shouldConnectLocally)
                 else
                 {
                     arguments.Append($"+{RuntimeConfigNames.Environment} {RuntimeConfigDefaults.CloudEnvironment} ");
+
+                    var gdkToolsConfig = GdkToolsConfiguration.GetOrCreateInstance();
+
+                    // Return error if no DevAuthToken is set AND fails to generate new DevAuthToken.
+                    if (!PlayerPrefs.HasKey(RuntimeConfigNames.DevAuthTokenKey) && !DevAuthTokenUtils.TryGenerate())
+                    {
+                        Debug.LogError("Failed to generate a Dev Auth Token to launch mobile client.");
+                        return;
+                    }
+
+                    // Prints a warning if DevAuthToken.txt does not exist, if GdkToolsConfiguration is configured
+                    // to save a DAT to file.
+                    if (gdkToolsConfig.SaveDevAuthTokenToFile && !File.Exists(gdkToolsConfig.DevAuthTokenFilepath))
+                    {
+                        Debug.LogWarning("Launching Android client without a DevAuthToken.txt asset.");
```

Comment:
would remove this check (or rephrase it quite a bit). cause the text file might have existed when the worker was built and for some reason deleted later. Additionally, we throw an error when starting the app, if we can't find the dev auth token. that should be enough